### PR TITLE
feat: subscriptions improvements

### PR DESCRIPTION
**Pull request by dinhani-cw**
* Using Subscription ID as String instead of number to be compatible with Geth client.
* Fix log filter:
  * It can receive multiple address.
  * It can received multiple combination of topics.

### CODE REVIEW

File: src/eth/primitives/log_filter_input.rs [hunk 1]
```diff
@@ -21,11 +28,13 @@ pub struct LogFilterInput {
...
 
-    #[serde(default)]
-    pub address: Option<Address>,
+    #[serde(rename = "address", default)]
+    #[serde_as(deserialize_as = "OneOrMany<_, PreferMany>")]
+    pub address: Vec<Address>,
 
-    #[serde(default)]
-    pub topics: Vec<Option<LogTopic>>,
+    #[serde(rename = "topics", default)]
```
Comment (carneiro-cw) [inline src/eth/primitives/log_filter_input.rs:35]: Another redundant rename

File: src/eth/primitives/log_filter_input.rs [hunk 2]
```diff
@@ -21,11 +28,13 @@ pub struct LogFilterInput {
     #[serde(rename = "blockHash", default)]
     pub block_hash: Option<Hash>,
 
-    #[serde(default)]
-    pub address: Option<Address>,
+    #[serde(rename = "address", default)]
```
Comment (carneiro-cw) [inline src/eth/primitives/log_filter_input.rs:31]: Isn't the renaming redundant here?

**Review by carneiro-cw — COMMENTED** [review]
Only thing I noticed out of the ordinary was those two redundant renames. I imagine there is a reason for it since previously they were not present.

**Comment by dinhani-cw** [comment]
@carneiro-cw 
The two new renames are not strictly necessary. 

They are an additional protection against someone renaming one of these fields without knowing they are part of an specification and breaking things unknowingly.

The two modified fields became `Vec` in this PR, so someone may be tempted to rename `address` to `addresses`. Even if the field is renamed, the JSON parsing will not break.

File: src/eth/primitives/log_filter_input.rs (hunk 1 above)
Comment (dinhani-cw) [reply src/eth/primitives/log_filter_input.rs:35]: Answer below.

File: src/eth/primitives/log_filter_input.rs (hunk 2 above)
Comment (dinhani-cw) [reply src/eth/primitives/log_filter_input.rs:31]: See below.

**Pull request merged**